beacon-chain/db/kv/kv.go

```diff
@@ -216,6 +222,23 @@ func (s *Store) DatabasePath() string {
 	return s.databasePath
 }
 
+func (s *Store) checkNeedsResync() error {
+	return s.db.View(func(tx *bolt.Tx) error {
+		bkt := tx.Bucket(migrationsBucket)
+		hasDisabledFeature := !features.Get().EnableOnlyBlindedBeaconBlocks
+		if hasDisabledFeature && bkt.Get(migrationBlindedBeaconBlocksKey) != nil {
+			log.Warnf(
+				"You have disabled the flag %s, and your node must resync to ensure your "+
+					"database is compatible. If you do not want to resync, please re-enable the %s flag",
+				features.EnableOnlyBlindedBeaconBlocks.Name,
+				features.EnableOnlyBlindedBeaconBlocks.Name,
+			)
+		}
+		return nil
+	})
```

Comment:
What do you think about surfacing this as an error up the process and exit the node if this error occurs?